### ISSUE TITLE
CACHEDIR.TAG and security

### ISSUE THREAD

**Issue by tleydxdy** [restic/restic#3604]
According to "Security Considerations" part of the [Cache Directory Tagging Specification](https://bford.info/cachedir/) which restic uses for the `--exclude-caches` flag, it seems advisable to block and ask to confirm the list of excluded cache directories (or new ones since the last backup at least) so user don't get surprises when some directory was not backed up (In my case, I can imagine user putting "cache directories" in their nextcloud, and I would like to keep that backed up, or at least noticed it and tell user to remove it or something)

**Comment by MichaelEischer** [comment]
The backup command does not support any kind of user interaction.

Why even use the `--exclude-cache` when you're worried about unwanted exclusions? You could instead keep track of `CACHEDIR.TAG`s by using `find` to determine the relevant folders, then use `diff` to check for changes and just feed restic an `--exclude-file` with curated content.

**Comment by tleydxdy** [comment]
sure, I could also just tar up my system to back up. I use restic because it's a good tool, I'm just suggesting this particular function could lead to data loss so maybe there need to be more warnings or reworked abit (perhaps exactly like you said, make it generate a exclude.txt file instead)

**Comment by jniggemann** [comment]
> I'm just suggesting this particular function could lead to data loss so maybe there need to be more warnings or reworked abit

Well, in your threat model, the user could drop a CACHEDIR.TAG in a folder, leading to it not being backed up.
Did you realize that the user could also just delete the folder right away?

**Comment by tleydxdy** [comment]
I thought I was being quite clear with
> (In my case, I can imagine user putting "cache directories" in their nextcloud, and I would like to keep that backed up, or at least noticed it and tell user to remove it or something)

**Comment by aawsome** [comment]
@tleydxdy: No one forces you to use `--exclude-caches`

- if you want to exclude a priori known cache directories, exclude them using `--exclude` or `--exclude-file`
- if you want to exclude directories depending on contents of the directory, use `--exclude-caches` or  `--exclude-if-present`

In the first case, nobody can toggle exclude rules even when modifying some content.
In the second case, anyone who can change the content can force dirs not to be backup'ed. 

So, don't understand your problem. For me, it just looks like the first case applies to your use case.

**Comment by tleydxdy** [comment]
@aawsome I agree with the things you said 100%, I just mean the flag carries some not so obvious risks and it would be useful to give a hint to the user to avoid mistakes.
